src/main/java/humi/TaskManager.java

```diff
@@ -0,0 +1,142 @@
+package humi;
+
+import java.io.File;
+import java.util.ArrayList;
+
+public class TaskManager {
+    //public static final int MAX_TASK = 100;
+    ArrayList<Task> taskList;
+    public static int taskCount;
+
+    TaskManager() {
+        taskList = new ArrayList<>();
+        taskCount = 0;
+    }
+
+    // add task from text file
+    public void addTask(String inputString) {
+        String[] taskComponents = inputString.split("/");
+        String taskType = taskComponents[0];
+        boolean isDone = (taskComponents[1].equals("1"));
+        String description = taskComponents[2];
+
+        if (taskType.equals("T")) {
+            taskList.add(new TodoTask(description, isDone));
+            taskCount++;
+        } else if (taskType.equals("D")) {
+            String deadline = taskComponents[3];
+            taskList.add(new DeadlineTask(description, deadline, isDone));
+            taskCount++;
+        } else if (taskType.equals("E")) {
+            String startDate = taskComponents[3];
+            String endDate = taskComponents[4];
+            taskList.add(new EventTask(description, startDate, endDate, isDone));
+            taskCount++;
+        } else {
+            System.out.println("Invalid text, cannot add new task.");
+        }
+    }
+
+    public void handleCommand(String command) {
+        try {
+            if (command.equals("list")) {
+                handleList();
+            } else if (command.startsWith("todo")) {
+                handleTodo(command);
+                String text = FileProcessor.convertCommandToText(command);
+                FileProcessor.appendTextContent(text);
```

Comment:
Instead of writing the code over and over again, consider abstracting them to a method to handle the task and invoke the method here instead.